Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -82,72 +100,186 @@ public void OnError(Exception error)
         }
 
         /// <summary>
-        /// Get the DependencyTelemetry objects that are still waiting for a response from the dependency. This will most likely only be used for testing purposes.
+        /// Enables HttpClient instrumentation with activity. It is never called and allows Microsoft.Extensions.DiagnosticAdapter to know that event is enabled.
         /// </summary>
-        internal IEnumerable<DependencyTelemetry> PendingDependencyTelemetry
+        [DiagnosticName("System.Net.Http.HttpRequestOut")]
+        public void IsActivityEnabled()
         {
-            get { return pendingTelemetry.Values; }
         }
 
         /// <summary>
-        /// Diagnostic event handler method for 'System.Net.Http.Request' event.
+        /// Handler for Exception event, it is sent when request processing cause an exception (e.g. because of DNS or network issues)
+        /// There will be anoth Stop event sent with null response.
         /// </summary>
-        [DiagnosticName("System.Net.Http.Request")]
-        public void OnRequest(HttpRequestMessage request, Guid loggingRequestId)
+        [DiagnosticName("System.Net.Http.Exception")]
+        public void OnException(Exception exception, HttpRequestMessage request)
         {
-            try
+            this.client.TrackException(exception);
```

Comment:
by associate you mean that dependency.ResultCode would have information about the exception?

The problem here is that HttpClient on .NET Desktop used to throw WebException that has a [WebExceptionStatus](https://msdn.microsoft.com/en-us/library/system.net.webexceptionstatus(v=vs.110).aspx)

On Core, there is no such thing. Instead there is WinHttpException (on windows) that does not have any status. There is only  a message like "The server name or address could not be resolved".
Do you think it's a good idea to have it in the resultCode and potentially group by it?

BTW, there is an association by root id (with correlation).